src/entities/EditorDocument/EditorDocument.spec.ts

```diff
@@ -0,0 +1,228 @@
+import { EditorDocument } from './index';
+import { BlockNode, BlockNodeName, createBlockNodeName } from '../BlockNode';
+
+const createBlock = ({ name, parent }: { name?: BlockNodeName, parent: EditorDocument }): BlockNode => {
+  return new BlockNode({
+    name: name || createBlockNodeName('block'),
+    parent,
+    children: {},
+  });
+};
+
+describe('EditorDocument', () => {
+  let document: EditorDocument;
+  let blocks: BlockNode[];
+
+  beforeEach(() => {
+    document = new EditorDocument({
+      children: [],
+      properties: {
+        readOnly: false,
+      },
+    });
+
+    const countOfBlocks = 3;
+
+    blocks = [];
+
+    for (let i = 0; i < countOfBlocks; i++) {
+      const block = createBlock({
+        parent: document,
+      });
+
+      document.addBlock(block);
+      blocks.push(block);
+    }
+  });
```

Comment:
lets create a document inside cases instead of beforeEach. Because "arrange" could have differences

Reply:
Why do I need to duplicate the same code? The specific document isn't needed in most cases.

Reply:
because test cases should not depend on each other. For example, the first test case mutates `document` variable. You can create `createDocumentMock(options)` helper to prevent code duplication